mat64/inner.go

```diff
@@ -29,26 +29,65 @@ func Inner(x []float64, A Matrix, y []float64) float64 {
 	switch b := A.(type) {
 	case RawSymmetricer:
 		bmat := b.RawSymmetric()
```

Comment:
I think the Triangular handling is for another PR. I'll add the Uplo check though.